util/metrics/src/service_metrics.rs

```diff
@@ -100,17 +119,25 @@ impl ServiceMetrics {
         )
     }
 
+    /// Count number of errors by method
     pub fn resp(&self, ctx: &RpcContext, success: bool) {
-        // The reason for counting everything here, instead of doing the
-        // if outside of the increment is that we could also compare
-        // number of responses to number of requests
         if let Some(name) = path_from_ctx(ctx) {
             self.num_error
                 .with_label_values(&[name.as_str()])
                 .inc_by(if success { 0 } else { 1 });
         }
     }
 
+    /// Count number of response codes by method
+    pub fn status_code(&self, ctx: &RpcContext, response_code: RpcStatusCode) {
+        if let Some(name) = path_from_ctx(ctx) {
+            self.num_status_code
+                .with_label_values(&[name.as_str(), response_code.to_string().as_str()])
+                .inc();
+        }
+    }
+
+    /// Track GRPC message size
```

Comment:
Similar comment as on the resp() call. I think this should indicate what the method is intended for so you know how to use it or extend it with additional metrics. Is this for counting metrics on messages that are sent or received?

Reply:
Yeah I actually agree, however, these methods were named as such prior to this and are used in a lot of different places which could cause larger breakage. If I change these method names, I'd want to do so as part of a larger refactor, which could be submitted as a separate PR and reviewed. At the moment though, I'm trying to prevent that type of scope creep by introducing a single method.